lib/storage/metadata/MetadataWrapper.js

```diff
@@ -54,13 +55,14 @@ function _parseListEntries(entries) {
  * @param {object[]} entries - Version or Content entries in a metadata listing
  * @param {string} entries[].key - metadata key
  * @param {string} entries[].value - stringified object metadata
+ * @param {function} parser - Parser for object data
  * @return {(object|Error)} - mapped array with parsed value or JSON parsing err
  */
-function parseListEntries(entries) {
+function parseListEntries(entries, parser) {
     // wrap private function in a try/catch clause
     // just in case JSON parsing throws an exception
     try {
-        return _parseListEntries(entries);
+        return parser(entries);
     } catch (e) {
         return e;
     }
```

Comment:
Metadata wrapper still has bucketclient for now. This is due to bucketclient importing Arsenal and creating a circular dependency yarn can not resolve. This is easily fixed in the move to Armory.